### PR TITLE
Search In Open Editors

### LINKED ISSUE #106790: Investigate predefined file filters search
In #20530 we've discussed custom file sets for search, and in the process the originally simple ask for a way to search in opened editors evolved into a large scale feature request involving extension contribution APIs, new UX, and updates to the search UI. 

This issue tracks scaling that back to **just searching in opened editors** in order to more quickly experiment with various ways of implementing the general feature. 

There are two primary proposals for how this might look: 

1. "Universe" Dropdown. This is what products like Dreamweaver use, where a dropdown contains a collection of "universes" of files to search: 
XyjNz

In this case, the dropdown would configure the "universe" of files to search, and the include and exclude filters would further narrow down files within that universe. 

In this model, one could search things like "all opened editors with .ts extension", but not things like "all opened editors and additionally all files in workspace with .ts extension" or "all opened editors and all test files" (assuming in the future extensions can contribute items)

2. Custom Include/Exclude Globs. This is what products like Sublime use:
image
(Note sublime surfaces these globs via a dropdown, but this is not the "universe" UX of above)
And would likely be surfaced through `@` blobs similar to the extension search:
image

In this case, each of these globs are entered directly into the include/exclude inputs, and they expand to some set of files/globs at searchtime.

In this model, one could search things like "all opened editors and additionally all files in workspace with .ts extension" and "all opened editors and also all test files", but not things like "all opened editors with .ts extension".

One note of consideration with this approach is that we sill would need to support searching in a directory literally named "`@open`", so some means of escaping or disabling the expansions is required. This could be done with a "use `@` expansions" toggle in the include input, similar to the existing "use global ignores" toggle in the exclude input: 
image

Open questions: 
1. Which model is more important to support? "This universe of files, filtered with these include/excludes", or "include/exclude this meta-glob in addition to the rest of my globs"? IMO the first seems more helpful, but I could be convinced either way.
2. Do these UI's necessarily force the above models?

### PR DESCRIPTION
**Pull request by JacksonKearl**
Work in progress. Needs some larger architectural changes to search to allow for requesting results from a single file. 

This PR fixes #106790

### CODE REVIEW

**Comment by roblourens** [comment]
Looks good, I think it would be fine to begin with something like this and we can keep discussing fancier solutions. In case you aren't aware, this will also be hit by https://github.com/microsoft/vscode/issues/62488 - only files that have been focused and loaded in this vscode session will be searched. Existing issue, but we may want to prioritize fixing that since this could make it more obvious.

File: src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts [hunk 1]
```diff
@@ -42,6 +42,9 @@ export class RipgrepTextSearchEngine {
 
 			const cwd = options.folder.fsPath;
 
+			console.log({ query, options, rgArgs, cwd });
```
Comment (roblourens) [inline src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: 👉

Comment (JacksonKearl) [reply src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: oops

Comment (JacksonKearl) [reply src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: Wait I removed that already, I think you're looking at an old version

Comment (roblourens) [reply src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts]: Ah I think the gh notification will just take to some random new commit which is not necessarily the latest one

**Pull request merged**